### ISSUE TITLE
Namespace break adjacent sibling combinator selector

### ISSUE THREAD

**Issue by slestang** [QuickBase/babel-plugin-styled-components-css-namespace#12]
First all thank for this plugin, is solved a very annoying CSS override issue between our 2 projects.

My issue is that:
```es6
const InfoContainer = styled.div`
  display: flex;
  & + & {
    margin-left: 0.5rem;
  }
`;
```

now generate the style ('spa' is our namespace):
```css
.sc-eHgmQL {}.spa .hVujAK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;} .spa .hVujAK + .spa .hVujAK{margin-left:0.5rem;}
```

With that style the margin-left is never applied because the selector should be `.spa .hVujAK + .hVujAK` instead of `.spa .hVujAK + .spa .hVujAK`

Is that fixable in babel-plugin-styled-components-css-namespace?

**Comment by nvanselow** [comment]
Thanks for the issue. We've seen this once in our codebase as well, although we don't often use that selector. We'll have to investigate, but it seems possible.

**Comment by nvanselow** [comment]
As a quick update, I've done a bit of digging and the issue appears to be the way `&` is interpreted once we add it to the top level selector. It looks like the plugin is correctly adding the class only to the top level, but that is later being picked up by the one's inside. I plan to do a bit more on this, but unfortunately it won't be a quick fix.

**Comment by stevenremot** [comment]
Hi, I got this problem too in styled components like this:

```js
const MyFirstExample = styled.div`
  input:checked + & { ... }
`

const MySecondExample = styled.div`
  .invalid & { ... }
`
```

I have the feeling that solving this problem requires much more than just wrapping the whole CSS code with `&& {` / `.prefix & {`, as it's done today. I think the plugin should be able to detect nested rules and apply the prefix for each selector, that's a whole new level of parsing 😄 

Do you have a simpler idea in your mind?

**Comment by nvanselow** [comment]
Fixed by #15 
Please use the new rawCssNamespace option.
https://github.com/QuickBase/babel-plugin-styled-components-css-namespace#multiple-namespaces

Thanks to @daniloster for the excellent approach and fix!
